src/App.tsx

```diff
@@ -17,18 +44,33 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                selectedCategory={selectedCategory}
+                setSelectedCategory={setSelectedCategory}
+                inputValue={inputValue}
+                setInputValue={setInputValue}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
```

Comment:
Ensure that the `Loader` component is displayed correctly when fetching data from the API. The task requires showing the `Loader` while waiting for any data from the server .